src/ansiblelint/yaml_utils.py

```diff
@@ -598,6 +598,36 @@ def choose_scalar_style(self) -> Any:
             return "'"
         return self.preferred_quote
 
+    def increase_indent(
+        self,
+        flow: bool = False,  # noqa: FBT002
+        sequence: bool | None = None,
+        indentless: bool = False,  # noqa: FBT002
+    ) -> None:
+        self.indents.append(self.indent, sequence)
+        if self.indent is None:  # top level
+            if flow:
+                # self.indent = self.best_sequence_indent if self.indents.last_seq() else \
+                #              self.best_map_indent
+                # self.indent = self.best_sequence_indent
+                self.indent = self.requested_indent
+            else:
+                self.indent = 0
+        elif not indentless:
+            if self.indents.last_seq():
+                self.indent += self.best_sequence_indent  # - self.sequence_dash_offset
+                if sequence:
+                    self.indent -= self.sequence_dash_offset
```

Comment:
This is what's doing the squeezing. If the _last_ indentation it desired was for a sequence, and this one is too, then ignore the dash offset.

This is likely also the reason for the extra newlines. ruamel is getting confused about where in the file we are, possibly as a consequence of `self.indents` getting out of sync? `self.column` ends up referring to a place far beyond the column we are at in the file and inserts a newline to try to catch up.

This can all probably get smartened up, this is just the first version I produced after playing around with it that made valid YAML instead of a huge mess.